### ISSUE TITLE
Taking Quickstart one step further for newbies

### ISSUE THREAD

**Issue by serpent213** [numtide/nixos-anywhere#202]
Assuming *nixos-anywhere* is and will be an entry point to Nix(OS) for many users (like me), it would be great to extend the Quickstart guide by one step.

I had a pretty flawless experience getting NixOS to run, but now I'm a bit stuck. Would be lovely to have a little guide on how to move on from here... That is, from the *flake.nix* I used to setup the server to some kind of (flake based) configuration to install a package, let's say. Flakes still being experimental, makes it even more cumbersome, and it surely shouldn't be a documentation on flakes. Just a simple example setup, assuming the configuration will be held on the target server itself, to make it easy.

For example I'm wondering, can I use the setup flake as is on the target? Do I need to copy parts of it? The target at least needs to know about the boot loader configuration, right?

Probably I would be happy with a flake-less approach as well – I don't know yet; and I started the setup before the flake-less section was added to the docs.

Thank you!

**Comment by serpent213** [comment]
See also #151

**Comment by phaer** [comment]
> For example I'm wondering, can I use the setup flake as is on the target?  Do I need to copy parts of it?

Yes, you can use it as is - there's an example at the end of the quick start guide using `nixos-rebuild` and it's `--target-host` flag; which is probably the most conventional tooling.

You could also populate a configuration.nix with the same settings and modules as the config in your flake and rebuild that, but I don't think there's an automated way or even a detailed walk-through for beginners yet?

The myriad of options here make it hard to give good general guidance on how to deploy and maintain your NixOS systems in general, as nixos-anywhere is still just an installer after all :)


> The target at least needs to know about the boot loader configuration, right?

Generally yes, but that should be included in your flake if you successfully installed nixos? ("Generally" b.c. there's things like `boot.loader.external.enable` where this would not apply)

**Comment by serpent213** [comment]
> The myriad of options here make it hard to give good general guidance on how to deploy and maintain your NixOS systems in general, as nixos-anywhere is still just an installer after all :)

Fully agree, but a “how to install your first package” would be a very valuable help.

Thanks for your input, will have a look!

**Comment by laundmo** [comment]
Just adding in here that i'd also appreciate if something like "Common steps after the install is finished" was documented somewhere. I just, as a complete Nix and Nixos newbie, used nixos-anywhere to install nixos on a server. and promptly had to call in help to understand why i cant find my flake.nix/configuration.nix anywhere on the server... i had (wrongly) assumed nixos-anywhere would copy it over by default, as from other things i've been consuming i understood the normal workflow to be "edit the config on your device, nixos-rebuild, changes are now applied"

**Comment by sedlund** [comment]
@laundmo 
> something like "Common steps after the install is finished" was documented somewhere. 

https://github.com/nix-community/nixos-anywhere/blob/main/docs/quickstart.md#finished

did you miss this, or did you not find it helpful enough?  I would suggest adding a PR to clarify if it further to your expectations.

**Comment by laundmo** [comment]
> did you miss this, or did you not find it helpful enough? I would suggest adding a PR to clarify if it further to your expectations.

The latter, i did read it but it never mentioned that by default it won't copy over the files. i didn't even know it was possible to run nixos without its configuration existing, i just thought using the rebuild command with a explicit location must like, overwrite the local one or something.

i might write a PR once i'm just a bit more sure of... everything
